test/powershell/engine/Api/ProxyCommand.Tests.ps1

```diff
@@ -0,0 +1,164 @@
+            }
+            $texts -join ""
+        }
+
+        function ProxyTest {
+            [CmdletBinding(DefaultParameterSetName='Name')]
+            param (
+                [Parameter(ParameterSetName="Name", Mandatory=$true)]
+                [ValidateSet("Orange", "Apple")]
+                [string] $Name,
+
+                [Parameter(ParameterSetName="Id", Mandatory=$true)]
+                [ValidateRange(1,5)]
+                [int] $Id,
+
+                [Parameter(ValueFromPipeline)]
+                [string] $Message
+            )
+
+            DynamicParam {
+                $ParamAttrib  = [parameter]::new()
+                $ParamAttrib.Mandatory  = $true
+
+                $AttribColl = [Collection[System.Attribute]]::new()
+                $AttribColl.Add($ParamAttrib)
+
+                $RuntimeParam  = [RuntimeDefinedParameter]::new('LastName', [string], $AttribColl)
+                $RuntimeParamDic  = [RuntimeDefinedParameterDictionary]::new()
+                $RuntimeParamDic.Add('LastName',  $RuntimeParam)
+                return  $RuntimeParamDic
+            }
+
+            Begin {
+                $AllMessages = @()
+                if ($PSCmdlet.ParameterSetName -eq "Name") {
+                    $MyString = $Name, $PSBoundParameters['LastName'] -join ","
+                } else {
+                    $MyString = $Id, $PSBoundParameters['LastName'] -join ","
+                }
+            }
+
+            Process {
+                if ($Message) {
+                    $AllMessages += $Message
+                }
+            }
+
+            End {
+                $MyString + " - " + ($AllMessages -join ";")
+            }
+        }
+    }
+
+    It "Test ProxyCommand.GetHelpComments" {
+        $helpObj = Get-Help Get-Alias -Full
+        $helpContent = [System.Management.Automation.ProxyCommand]::GetHelpComments($helpObj)
+
+        $funcBody = @"
+    <#
+    {0}
+    #>
+
+    param({1})
+"@
+        $params = $helpObj.parameters.parameter
+        $paramString = '${0}' -f $params[0].name
+        for ($i = 1; $i -lt $params.Length; $i++) {
+            $paramString += ',${0}' -f $params[$i].name
+        }
+
+        $bodyToUse = $funcBody -f $helpContent, $paramString
+        $bodySB = [scriptblock]::Create($bodyToUse)
+        Set-Item -Path function:\TestHelpComment -Value $bodySB
+        $newHelpObj = Get-Help TestHelpComment -Full
+
+        $helpObj.Synopsis | Should Be $newHelpObj.Synopsis
+        $oldDespText = GetSectionText $helpObj.description
+        $newDespText = GetSectionText $newHelpObj.description
+        $oldDespText | Should Be $newDespText
+
+        $oldParameters = @($helpObj.parameters.parameter)
+        $newParameters = @($newHelpObj.parameters.parameter)
+        $oldParameters.Length | Should Be $newParameters.Length
+        for ($i = 0; $i -lt $oldParameters.Length; $i++) {
+            $oldParameters[$i].name | Should Be $newParameters[$i].name
```

Comment:
Do we really want report every value? Can we use `Compare-Object` to be fast?

Reply:
Replaced with `$oldParameters.name -join "," | Should Be ($newParameters.name -join ",")`